pkg/planner/core/initialize.go

```diff
@@ -249,12 +249,27 @@ func (p PhysicalUnionScan) Init(ctx base.PlanContext, stats *property.StatsInfo,
 	return &p
 }
 
+func (p *PhysicalIndexLookUpReader) adjustReadReqType(ctx base.PlanContext) {
+	if p.StoreType == kv.TiFlash {
+		_, ok := p.tablePlan.(*PhysicalExchangeSender)
+		if ok {
+			p.ReadReqType = MPP
+			return
+		}
+		// When allow batch cop is 2, every query uses batch cop.
+		if ctx.GetSessionVars().AllowBatchCop == 2 {
```

Comment:
Maybe just set `p.ReadReqType` to `BatchCop`? or if `AllowBatchCop !=0` then set `p.ReadReqType` to `BatchCop`

Reply:
addressed.